pkg/data/egress_lists/egress_lists.go

```diff
@@ -31,12 +24,60 @@ var templateGCPClassic string
 //go:embed aws-hcp-zeroegress.yaml
 var templateAWSHCPZeroEgress string
 
-func GetLocalEgressList(platformType cloud.Platform) (string, error) {
-	if !platformType.IsValid() {
-		fmt.Printf("platform type %s is invalid", platformType)
+type githubReposClient interface {
+	GetContents(ctx context.Context, owner, repo, path string, opts *github.RepositoryContentGetOptions) (fileContent *github.RepositoryContent, directoryContent []*github.RepositoryContent, resp *github.Response, err error)
+}
+
+type Generator struct {
+	PlatformType      cloud.Platform
+	Variables         map[string]string
+	logger            logging.Logger
+	githubReposClient githubReposClient
+}
```

Comment:
Could we add a docstring here, and perhaps change the name to something less generic? `EgressListGenerator` perhaps?

Reply:
I chose the name `Generator` so it lines up nicely with the package name. `egress_lists.Generator` is pretty clear in the code, versus `egress_lists.EgressListGenerator` is a bit verbose, but I am happy to change it if you feel strongly

Reply:
Ahh right, I always forget that everything gets prefixed with the package name. Nope, that makes perfect sense. Just slap on some comments for the type and its fields and we're good